lib/improver/utilities/cube_metadata.py

```diff
@@ -275,59 +283,120 @@ def update_attribute(cube, attribute_name, changes, warnings_on=False):
     return result
 
 
+def update_cell_methods(cube, cell_method_definition):
+    """Update cell methods. If the cell method defined in the
+    cell_method_definition matches a cell method that already exists within
+    the cube, then the cell method is removed. If the cell method defined
+    in the cell method definition does not match a cell method within the
+    cube, then the cell method is added.
+
+    The cube will be modified in-place.
+
+    Args:
```

Comment:
As discussed, we've now added an `action` keyword to the specification of the cell methods, so that it is clear whether a cell method is to be added or deleted, so as to avoid surprise behaviour.

Reply:
Thanks